2_run/src/run_glm3.R

```diff
@@ -22,6 +29,25 @@ extract_glm_output <- function(nc_filepath, nml_obj, export_fl) {
   ice_data <- glmtools::get_var(nc_filepath, var_name = 'hice') %>%
     mutate(ice = hice > 0, time = as.Date(lubridate::ceiling_date(DateTime, ' days'))) %>%
     select(-DateTime)
+  
+  # As a further check of whether or not the model run was successful, 
+  # check if # layers = 1 at any point during the simulation, and 
+  # return an error message if so
+  min_layers <- min(layers_data$n_layers)
+  if (min_layers == 1) {
+    error_message <- 'Number of layers drops to 1'
+    
+    # also check if max ice thickness exceeds lake depth, since that
+    # is sometimes the reason that the number of layers drops to 1
+    max_hice <- max(ice_data$hice)
+    
+    if (max_hice > lake_depth) {
```

Comment:
There weren't [here](https://github.com/USGS-R/lake-temperature-process-models/discussions/45#discussioncomment-2634481), and I believe that when ice formation is so extreme it will always drop to 1 layer, since so much water is used for the ice formation. But if we want to be extra sure I'm happy to add an additional check for ice thickness alone.

Reply:
Oh and it doesn't only update the error message, it also triggers an error [here](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L128-L130), since without this additional code, this function would run successfully and [return `NA` for `extraction_error`](https://github.com/USGS-R/lake-temperature-process-models/blob/main/2_run/src/run_glm3.R#L117).

Reply:
Right, sorry, I meant the additional ice check rather than just layers going to 1 only updates the error. I am certainly not an expert in what the scenarios are that GLM might produce. Just was wondering if we should have `if (max_hice > lake_depth)` not nested within the min layers. If you don't think that scenario would come up, let's leave how it is!

Reply:
Okay I added it in just in case that scenario does occur - I'd hate to not capture it and it's a quick change. I'm still preserving the ability for the error message to mention both layers and ice thickness if both cases are true, since I'd like to record that information for our reference if we need it.